botorch/test_functions/multi_objective.py

```diff
@@ -601,3 +606,54 @@ def evaluate_slack_true(self, X: Tensor) -> Tensor:
         min1 = (term1 + term2).min(dim=-1).values
         min2 = ((f_X - 1 / math.sqrt(f_X.shape[-1])).pow(2) - self._r ** 2).sum(dim=-1)
         return -torch.min(min1, min2).unsqueeze(-1)
+
+
+class OSY(MultiObjectiveTestProblem, ConstrainedBaseTestProblem):
+    r"""
+    The OSY test problem from [Oszycka1995]_.
+    Implementation from
+    https://github.com/msu-coinlab/pymoo/blob/master/pymoo/problems/multi/osy.py
+    Note that this implementation assumes minimization, so please choose negate=True.
+    """
+
+    dim = 6
+    num_constraints = 6
+    num_objectives = 2
+    _bounds = [
+        (0.0, 10.0),
+        (0.0, 10.0),
+        (1.0, 5.0),
+        (0.0, 6.0),
+        (1.0, 5.0),
+        (0.0, 10.0),
+    ]
+    _ref_point = [-75.0, 75.0]
+
+    def __init__(
+        self,
+        dim: int = 6,
+        num_objectives: int = 2,
```

Comment:
looks like this is always in dim 6 with 2 objectives? Let's remove these args then (these are for test functions where one can select the dimensionality).

Reply:
In fact, you can just omit the constructor altogether then since the superclass constructor will do all you need

Reply:
I originally built the constructor to have negate=True as the default option, but that was breaking the unit tests.